zeq/event.cpp

```diff
@@ -59,4 +54,14 @@ const flatbuffers::Parser& Event::getParser() const
     return _impl->parser;
 }
 
+void Event::setData( const void* data, const size_t size )
+{
+    _impl->setData( data, size );
+}
+
+void Event::setData( const ConstByteArray& data, const size_t size )
```

Comment:
Because one takes the ownership of the pointer (through a shared pointer) and the other one doesn't. I moved the documentation to zeq/event.h hoping it would be clearer.
An alternative is to provide just the smart pointer version and let the caller decide by providing a custom deallocator.